### ISSUE TITLE
GLA Battle Bus will enter bunkered mode at 50% health left when shot with Toxin and Radiation weapons

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsGamePatch#761]
GLA Battle Bus will enter bunkered mode at 50% health left when shot with Toxin and Radiation weapons.

https://user-images.githubusercontent.com/4720891/181082349-94efc7d9-2476-4a30-8f09-97cd5da44cdd.mp4

**Comment by xezon** [comment]
hanfield:
> most likely something for thyme to fix

**Comment by ImTimK** [comment]
Well I agree it's a bad bug, fixing it however will make Battle Busses so much harder to kill though and whereas Nuke for example had a chance before, it now would be defenseless until Nuke Migs (which trade expensively vs Gamma Busses). 

Sure you can say other balance improvements of the patch will make up for it, it's still 2 steps back.

**Comment by MTKing4** [comment]
Well we can reduce its hp then, that way all chinas will have better time against it and not just a nuke privilege.

**Comment by ImTimK** [comment]
> Well we can reduce its hp then, that way all chinas will have better time against it and not just a nuke privilege.

Other China's do already have a counter though, BN Migs. Gamma Busses are still a problem because of the unlimited RPG range, ECM's sometimes can save them, but it's rare.

**Comment by MTKing4** [comment]
True, in the case of air units other Chinas have an edge, but land units would be equal with this change, so maybe change how much damage an unupgraded nuke mig can do to a bus.

**Comment by xezon** [comment]
Setting POISON To 0% or 100% in Armor BattleBusTruckArmor, BattleBusStructureArmorTough does not affect this bug.

**Comment by xezon** [comment]
Issue appears to originate from the `Body = UndeadBody` module. Swapping that for a regular `ActiveBody` turns the Bus into a regular death body and the issue is gone.

**Comment by xezon** [comment]
Tweaking the `SecondLifeMaxHealth` value to be equal or lower `MaxHealth` makes no difference. I suspect there is math error in UndeadBody code.

**Comment by OccultGuy** [comment]
I found a mitigation for this. Very simply change the HazardFieldCoreWeapon to do less damage - like 10 and in HazardousMaterialArmor set HAZARD_CLEANUP to 2000% as an offset.
The undead body messes up at 50% hp and seems to apply all damage from the same source, even if not applicable. Since the HazardFieldCoreWeapon normally does 200 damage, half of the hp of a Battle Bus, it switches to second life as it is destroyed.

Now, this changes some matchups, but it's adjustable, since the 10 damage still get applied.
Nuke having trouble with the Battle Bus in transition from mid to lategame is good, since its one of the few drawbacks he has from being 100% China Vanilla, but better.
Gamma Busses do 480 dps, pre AP Rockets - very OP, but its not the fault of the bus itself.

**Comment by Float1ngFree** [comment]
> The undead body ~~messes up at 50% hp and~~ seems to apply all damage from the same source, even if not applicable.

This!
Can confirm that reducing the hazard cleaning damage delays the premature destruction. The additional damage is not accounted for until it exceeds the current health amount.
